src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/DuplicateWaysCheck.java

```diff
@@ -0,0 +1,115 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.Segment;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.AreaTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * This check looks for partially or completely duplicated Ways via Edges.
+ *
+ * @author savannahostrowski
+ */
+public class DuplicateWaysCheck extends BaseCheck
+{
+    // You can use serialver to regenerate the serial UID.
+    private static final long serialVersionUID = 1L;
+    public static final String DUPLICATE_EDGE_INSTRUCTIONS = "This way, {0,number,#}, "
+            + "has at least one duplicate segment.";
+
+    public static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(DUPLICATE_EDGE_INSTRUCTIONS);
+    public static final int ZERO_LENGTH = 0;
+    public static final String AREA_KEY = AreaTag.KEY;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public DuplicateWaysCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge
+                // Check to see that the Edge is a master Edge
+                && ((Edge) object).isMasterEdge()
+                // Check to see that the edge has not already been seen
+                && !this.isFlagged(((Edge) object).getMasterEdgeIdentifier())
```

Comment:
If this condition is being checked, then `((Edge) object).isMasterEdge()` must be true, as well as `object instanceof Edge`. So you don't have to cast `object` here. Just get it's identifer and check if it is flagged or not.